src/WebJobs.Script.WebHost/Controllers/FunctionsController.cs

```diff
@@ -59,17 +64,58 @@ public FunctionsController(WebScriptHostManager scriptHostManager, WebHookReceiv
             return await _scriptHostManager.HttpRequestManager.ProcessRequestAsync(request, processRequestHandler, cancellationToken);
         }
 
-        public static async Task<AuthorizationLevel> DetermineAuthorizationLevelAsync(HttpRequestMessage request, FunctionDescriptor function, IDependencyResolver resolver)
+        public static async Task<AuthorizationLevel> DetermineAuthorizationLevelAsync(HttpRequestMessage request, FunctionDescriptor function, ScriptSettingsManager settingsManager, IDependencyResolver resolver)
         {
+            // first check if we're authorized via key
             var secretManager = resolver.GetService<ISecretManager>();
             var authorizationResult = await AuthorizationLevelAttribute.GetAuthorizationResultAsync(request, secretManager, functionName: function.Name);
             var authorizationLevel = authorizationResult.AuthorizationLevel;
+
+            if (authorizationLevel > AuthorizationLevel.Anonymous)
+            {
+                // add a new identity to the principal representing
+                // the key authentication result
+                AddKeyAuthorizationIdentity(ClaimsPrincipal.Current, authorizationResult);
+            }
+
+            // See if we're we're authorized at the User level (EasyAuth)
```

Comment:
We want to determine a single auth level for the request but must deal with the possibility that a request might include both a key as well as EasyAuth credentials.